app/Http/Controllers/UserController.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Requests\UserRequest;
+use App\User;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\DB;
+
+class UserController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Contracts\Foundation\Application|\Illuminate\Contracts\View\Factory|\Illuminate\View\View
+     */
+    public function index()
+    {
+        $users =  User::paginate(config('paginates.ten'));
+
+        return view('admin.user_management.show_user', compact('users'));
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        return view('admin.user_management.create_user');
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(UserRequest $request)
+    {
+        User::create($request->all());
+
+        return redirect()->route('users.index')->with('success', trans('message.created'));
+    }
+
+    /**
+     * Display the specified resource.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function show($id)
+    {
+
+    }
+
+    /**
+     * Show the form for editing the specified resource.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function edit($id)
+    {
+        $user = User::findOrFail($id);
+
+        return view('admin.user_management.update_user', compact('user'));
```

Comment:
Hàm findOrFail này hoạt động như thế nào ?

Reply:
findOrFail sẽ tìm trong bảng users trả về bản ghi có id = $id nếu tìm thấy, nếu không thì nó sẽ throw ra một cái error ạ

Reply:
Vậy như trên, throw ra error thì vẫn trả về view à ?

Reply:
sẽ trả về trang 404 not found ạ

Reply:
em dùng try catch để custom lại khi gặp error nhé